client/src/components/AddCard/AddCardForm/AddCardForm.tsx

```diff
@@ -0,0 +1,122 @@
+import { useState, useEffect } from 'react';
+import Button from '@material-ui/core/Button';
+import Box from '@material-ui/core/Box';
+import Grid from '@material-ui/core/Grid';
+import { Formik, FormikHelpers } from 'formik';
+import * as Yup from 'yup';
+import Typography from '@material-ui/core/Typography';
+import useStyles from './useStyles';
+import { useAuth } from '../../../context/useAuthContext';
+import { CircularProgress } from '@material-ui/core';
+import { addCard } from '../../../helpers/APICalls/addCard';
+import { getAllPaymentMethods } from '../../../helpers/APICalls/getAllPaymentMethods';
+import { useSnackBar } from '../../../context/useSnackbarContext';
+import { useHistory } from 'react-router-dom';
+import CreditCard from './CreditCard/CreditCard';
+
+interface Props {
+  handleSubmit: (
+    {
+      paymentMethod,
+    }: {
+      paymentMethod: string;
+    },
+    {
+      setStatus,
+      setSubmitting,
+    }: FormikHelpers<{
+      paymentMethod: string;
+    }>,
+  ) => void;
+}
+
+export default function AddCard({ handleSubmit }: Props): JSX.Element {
+  const [paymentMethods, setPaymentMethods] = useState<any>([]);
+  const { loggedInUser } = useAuth();
+  const { updateSnackBarMessage } = useSnackBar();
+  const history = useHistory();
+
+  useEffect(() => {
+    if (loggedInUser != undefined && loggedInUser) {
```

Comment:
The same comment as with testing for truthiness above

Reply:
👍🏻Done